lib/brain/redis.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+require 'socket'
+require 'logger'
+require 'json'
+require 'yaml'
+require 'docker'
+require 'redis'
+
+# Brain Class
+# Its main functions are to:
+#  1. Create Redis Container
+#     - Local Docker Install or DIND depending on Setup
+#     - Validates if a previous container is running and replaces it
+#     - Mounts Path to DIND Host or Localhost for Redis AOF file (Data Persistance)
+#  2. Create Redis Client Access
+#     - URL For Redis determined by IP Comparison (Checking for Compose Environment)
+#  3. Enables Access to Brain
+#     - Query brain for specific key (i.e. Plugin User)
+#     - Query brain for specific hash (i.e. Plugin)
+#     - Delete Key from Brain (i.e. Plugin User)
+#     - Save Key/Value to Brain (i.e. Plugin User Bob )
+class Brain
+  def initialize(settings)
+    @logger = Logger.new(STDOUT)
+    @logger.level = settings.logger_level
```

Comment:
I believe we should log an issue to come back to for getting the logger set at a more global level.  Similar to some of the thoughts here: http://stackoverflow.com/questions/5995854/logging-in-sinatra